lib/rubygems/commands/owner_command.rb

```diff
@@ -87,6 +88,16 @@ def manage_owners method, name, owners
         response = rubygems_api_request method, "api/v1/gems/#{name}/owners" do |request|
           request.set_form_data 'email' => owner
           request.add_field "Authorization", api_key
+          request.add_field "OTP", options[:mfa] if options[:mfa]
+        end
+
+        if need_mfa? response
+          check_mfa
+          response = rubygems_api_request method, "api/v1/gems/#{name}/owners" do |request|
+            request.set_form_data 'email' => owner
+            request.add_field "Authorization", api_key
+            request.add_field "OTP", options[:mfa]
+          end
```

Comment:
Please don't duplicate code. Make a method out of common code where otp is optional argument.
Also, I agree with @colby-swandale 
> the code would be clearer if you stick with just the one name.

`"OTP", options[:mfa]` :no_entry_sign: 
Please use `otp` for everything.